app/utils.py

```diff
@@ -1177,7 +1178,16 @@ class XsrfTool(object):
     TOKEN_EXPIRATION_TIME = 60 * 60 * 4
 
     def __init__(self):
-        self._key = config.get('xsrf_token_key').encode('utf-8')
+        configured_key = config.get('xsrf_token_key')
+        if configured_key:
+            # config.get returns unicode, but hmac is going to want a str
+            self._key = configured_key.encode('utf-8')
+        else:
+            configured_key = ''.join([
+                random.choice(string.letters + string.digits)
+                for i in range(20)])
```

Comment:
depending on how fancy you want to be, I'd suggest:

option a: no [] and no i:
ALPHANUMERIC = string.letters + string.digits
def __init....
configured_key = ''.join(
                random.choice(ALPHANUMERIC)
                for _ in range(20))

option b: no for loop and yet another library:
ALPHANUMERIC = list(string.letters + string.digits)
def __init....
  configured_key = ''.join(numpy.random.choice(ALPHANUMERIC, 20, replace=True))

Reply:
Thanks, good points! We're not using numpy for anything else so I went with option A.